### ISSUE TITLE
Silently logged out

### ISSUE THREAD

**Issue by MPvHarmelen** [jotta/jotta-cli-issues#99]
Make sure you are running the latest version of jotta-cli before reporting an issue.

**jotta-cli release (`jotta-cli version`):**
```
-------------------------------------------
jottad executable : /usr/bin/jottad
jottad appdata    : /var/lib/jottad
jottad version    : 0.6.18626
jotta-cli version : 0.6.18626
-------------------------------------------
OK
```

**Description of problem:**
I frequently am logged out, without doing so explicitly. This happens silently and the only way to make sure my backups are running correctly is to run `jotta-cli status` and hope that it does not say `ERROR  Not logged in`.

After logging in again, I have to re-add and scan my backup folders.

**Expected:**
I expect to stay logged in if I don't run any commands.

**jotta-cli status (`jotta-cli status`):**
```
ERROR  Not logged in
```

**Relevant logs for the issue (`cat ~/.jottad/jottabackup.log`)**
(My log file is apparently stored under `/var/lib/jottad/jottabackup.log`.)
```
(very many more of the following three lines before this point in the log)
pid:860 2019/11/23 10:13:33 allocate: dc895c93f859ff6e6d30a3e625a673d5 (/jfs/ANONYMISED/.cache/mozilla/firefox/3z4rs993.default/cache2/entries/E5C8DE5977A05F8D382AD62152EAEE73FBEEDB62) => tokenrequest: performTokenRequest: Post https://id.jottacloud.com/auth/realms/jottacloud/protocol/openid-connect/token: dial tcp: lookup id.jottacloud.com: no such host
pid:860 2019/11/23 10:13:33 Error allocating tokenrequest: performTokenRequest: Post https://id.jottacloud.com/auth/realms/jottacloud/protocol/openid-connect/token: dial tcp: lookup id.jottacloud.com: no such host
pid:860 2019/11/23 10:13:33 Error uploading {"Name":"E5C8DE5977A05F8D382AD62152EAEE73FBEEDB62","Path":"/home/martin/.cache/mozilla/firefox/3z4rs993.default/cache2/entries/E5C8DE5977A05F8D382AD62152EAEE73FBEEDB62","Size":20253,"Modified":"2019-11-22T16:09:32.7427668+01:00","Created":"2019-11-22T16:09:32.7427668+01:00","Checksum":"","State":"Added"} => tokenrequest: performTokenRequest: Post https://id.jottacloud.com/auth/realms/jottacloud/protocol/openid-connect/token: dial tcp: lookup id.jottacloud.com: no such host
pid:860 2019/11/23 10:13:34 Scan of /home/martin completed in 4m35.788624247s. Found 419326 files 115.10GiB 
pid:860 2019/11/23 10:13:38 jottad.auth.err => tokenrequest: performTokenRequest: Post https://id.jottacloud.com/auth/realms/jottacloud/protocol/openid-connect/token: dial tcp: lookup id.jottacloud.com: no such host
pid:860 2019/11/23 10:13:45 ---------------
pid:860 2019/11/23 10:13:45 Daemon stopping
pid:860 2019/11/23 10:13:45 Stopping scheduled backup of /home/martin
pid:860 2019/11/23 10:13:45 Scheduler stopped
pid:860 2019/11/23 10:13:46 backup stopped
pid:860 2019/11/23 10:13:46 Database stopped
pid:860 2019/11/23 10:13:46 Downloader stopped
pid:860 2019/11/23 10:13:46 Daemon stopped
pid:858 2019/11/23 10:18:09 Starting /usr/bin/jottad version 0.6.18626
pid:858 2019/11/23 10:18:09 jottad.init
pid:858 2019/11/23 10:18:09 jottad.init.appdata => /var/lib/jottad
pid:858 2019/11/23 10:18:09 Unable to register device : Post https://api.jottacloud.com/auth/v1/register: dial tcp: lookup api.jottacloud.com: no such host
pid:858 2019/11/23 10:18:09 jottad.init.cfg.downloadratae => 0B
pid:858 2019/11/23 10:18:09 jottad.init.cfg.uploadrate => 0B
pid:858 2019/11/23 10:18:09 jottad.init.cfg.scaninterval =>  15m0s
pid:858 2019/11/23 10:18:09 jottad.init.user => no user logged in
pid:858 2019/11/23 10:18:13 jottad.init.done => ok
pid:858 2019/11/23 10:18:13 jottad.startup
pid:858 2019/11/23 10:18:13 jottad.startup => ok
pid:858 2019/11/23 10:18:13 api.grpc.active => 127.0.0.1:14443
pid:858 2019/11/23 10:18:13 Error checking for new versions Get https://repo.jotta.us/archives/VERSION: dial tcp: lookup repo.jotta.us: no such host
pid:858 2019/11/23 10:18:17 ---------------
pid:858 2019/11/23 10:18:17 Daemon stopping
pid:858 2019/11/23 10:18:17 Scheduler stopped
pid:858 2019/11/23 10:18:18 backup stopped
pid:858 2019/11/23 10:18:18 Database stopped
pid:858 2019/11/23 10:18:18 Downloader stopped
pid:858 2019/11/23 10:18:18 Daemon stopped
pid:806 2019/11/24 11:47:45 Starting /usr/bin/jottad version 0.6.18626
pid:806 2019/11/24 11:47:45 jottad.init
pid:806 2019/11/24 11:47:45 jottad.init.appdata => /var/lib/jottad
pid:806 2019/11/24 11:47:45 jottad.init.cfg.downloadratae => 0B
pid:806 2019/11/24 11:47:45 jottad.init.cfg.uploadrate => 0B
pid:806 2019/11/24 11:47:45 jottad.init.cfg.scaninterval =>  15m0s
pid:806 2019/11/24 11:47:45 Unable to register device : Post https://api.jottacloud.com/auth/v1/register: dial tcp: lookup api.jottacloud.com: no such host
pid:806 2019/11/24 11:47:46 jottad.init.user => no user logged in
pid:806 2019/11/24 11:47:46 Unable to register device : Post https://api.jottacloud.com/auth/v1/register: dial tcp: lookup api.jottacloud.com: no such host
pid:806 2019/11/24 11:47:46 jottad.init.done => ok
pid:806 2019/11/24 11:47:46 jottad.startup
pid:806 2019/11/24 11:47:46 jottad.startup.auth.patch
pid:806 2019/11/24 11:47:46 api.grpc.active => 127.0.0.1:14443
pid:806 2019/11/24 11:47:47 Unable to register device : Post https://api.jottacloud.com/auth/v1/register: dial tcp: lookup api.jottacloud.com: no such host
pid:806 2019/11/24 11:47:48 jottad.startup => ok
pid:806 2019/11/24 11:47:48 Error checking for new versions Get https://repo.jotta.us/archives/VERSION: dial tcp: lookup repo.jotta.us: no such host
pid:806 2019/11/24 11:47:48 Unable to register device : Post https://api.jottacloud.com/auth/v1/register: dial tcp: lookup api.jottacloud.com: no such host
(207 similar lines in between...)
pid:806 2019/11/24 11:51:52 Unable to register device : Post https://api.jottacloud.com/auth/v1/register: dial tcp: lookup api.jottacloud.com: no such host
pid:806 2019/11/24 11:52:23 Pausing for 10h0m0s. Resume @ 2019-11-24 21:52:23.913035506 +0100 CET m=+36246.776280377 
pid:806 2019/11/24 11:52:23 Scheduler stopped
pid:806 2019/11/24 13:39:45 paused: resuming in ~9h0m3.510836301s
pid:806 2019/11/24 15:35:09 Error checking for new versions Get https://repo.jotta.us/archives/VERSION: dial tcp: lookup repo.jotta.us: no such host
pid:806 2019/11/24 15:39:09 paused: resuming in ~8h0m3.511057411s
pid:806 2019/11/25 13:15:57 paused: resuming in ~7h0m3.510761295s
pid:806 2019/11/25 18:08:34 GET https://repo.jotta.us/archives/VERSION 200
pid:806 2019/11/25 18:12:33 paused: resuming in ~6h0m3.510733041s
pid:806 2019/11/26 09:30:41 paused: resuming in ~5h0m3.510747414s
pid:806 2019/11/26 10:53:50 GET https://repo.jotta.us/archives/VERSION 200
pid:806 2019/11/26 10:57:50 paused: resuming in ~4h0m3.510759282s
pid:806 2019/11/26 14:12:26 paused: resuming in ~3h0m3.510804954s
pid:806 2019/11/26 15:08:26 GET https://repo.jotta.us/archives/VERSION 200
pid:806 2019/11/26 15:12:26 paused: resuming in ~2h0m3.510930542s
pid:806 2019/11/27 12:27:32 paused: resuming in ~1h0m3.510822794s
pid:806 2019/11/27 13:18:32 paused: resuming in ~9m3.510731696s
pid:806 2019/11/27 13:19:32 paused: resuming in ~8m3.511137308s
pid:806 2019/11/27 13:20:32 paused: resuming in ~7m3.510630537s
pid:806 2019/11/27 13:21:32 paused: resuming in ~6m3.51075809s
pid:806 2019/11/27 13:22:32 paused: resuming in ~5m3.510711192s
pid:806 2019/11/27 13:23:32 paused: resuming in ~4m3.510745154s
pid:806 2019/11/27 13:23:32 GET https://repo.jotta.us/archives/VERSION 200
pid:806 2019/11/27 13:24:32 paused: resuming in ~3m3.510754829s
pid:806 2019/11/27 13:25:32 paused: resuming in ~2m3.510826268s
pid:806 2019/11/27 13:26:32 paused: resuming in ~1m3.510738828s
pid:806 2019/11/27 13:27:32 paused: resuming in ~3.510788058s
pid:806 2019/11/27 13:28:32 Resuming
pid:806 2019/11/27 16:16:59 Pausing for 1h0m0s. Resume @ 2019-11-27 17:16:59.792769617 +0100 CET m=+45408.532770000 
pid:806 2019/11/27 16:16:59 Scheduler stopped
pid:806 2019/11/27 16:40:16 GET https://repo.jotta.us/archives/VERSION 200
pid:806 2019/11/28 09:55:35 paused: resuming in ~9m45.267481538s
pid:806 2019/11/28 09:56:35 paused: resuming in ~8m45.267393128s
pid:806 2019/11/28 09:57:35 paused: resuming in ~7m45.267536981s
pid:806 2019/11/28 09:58:35 paused: resuming in ~6m45.267490193s
pid:806 2019/11/28 09:59:35 paused: resuming in ~5m45.267577094s
pid:806 2019/11/28 10:00:35 paused: resuming in ~4m45.267599451s
pid:806 2019/11/28 10:01:35 paused: resuming in ~3m45.267571159s
pid:806 2019/11/28 10:02:35 paused: resuming in ~2m45.267461873s
pid:806 2019/11/28 10:03:35 paused: resuming in ~1m45.267559211s
pid:806 2019/11/28 10:04:35 paused: resuming in ~45.267612054s
pid:806 2019/11/28 10:05:35 Resuming
```

**Traceback**
N.A.

**Additional info:**

**Comment by paalge** [comment]
I've experienced the same. 
```jotta-cli archive ``` ran for me but now files were shown on the web as i apparently was logged out. Don't know how this worked.

**Comment by roflmao** [comment]
Looks to me like you have DNS issues by the log you provided:

```
pid:806 2019/11/24 11:51:52 Unable to register device : Post https://api.jottacloud.com/auth/v1/register: dial tcp: lookup api.jottacloud.com: no such host
```

If the client cannot renew the token, it gets invalidated.

**Comment by MPvHarmelen** [comment]
It seems that the first line with such an error is this one:

```
pid:860 2019/11/23 10:13:38 jottad.auth.err => tokenrequest: performTokenRequest: Post https://id.jottacloud.com/auth/realms/jottacloud/protocol/openid-connect/token: dial tcp: lookup id.jottacloud.com: no such host
```

And it also seems a pity to immediately invalidate a token if renewal fails once due to network problems.

Are there security reasons to keep it this way, or would relaxing the invalidation rules solve this issue?

**Comment by Kimbsen** [comment]
I'm looking into this right now and a single dns lookup fail or other network related error should not log you out. Not sure how it caused a logout here.

But we do have a new release almost ready with some unrelated changes but also some improvements to auth. I think we'll get it out this week and hopefully that will fix your problem. Though i can't be entirely sure as i haven't found the underlying cause yet. I'll keep looking though.

**Comment by Kimbsen** [comment]
I have identified the issue, and it was indeed a single http fail that caused the logout. It occurs if  a token refresh is due before jottad starts and that http call fails during startup.

I'm working on the fix and I think it can be released tomorrow.

**Comment by Kimbsen** [comment]
Released 0.6.21799 which should fix this issue.
